File: javascript/platzi-course/creational/singleton/singleton.js
class Singleton {

    static instance = undefined;

    constructor(version) {
        this.version= version
    }

    static getInstance(version) {
        if (! Singleton.instance) {
            Singleton.instance = new Singleton(version);
        }
        return Singleton.instance;
    }


}

function appSingleton() {
    
    const singleton1= Singleton.getInstance('version01')
    const singleton2= Singleton.getInstance('version02')
    console.log(singleton1);
    console.log(singleton2);
    console.log(singleton1=== singleton2);
}

appSingleton();

File: design-pattern/java/design-pattern/src/main/java/ec/software/engineer/designpattern/creational/singleton/Singleton.java
package ec.software.engineer.designpattern.creational.singleton;

public class Singleton {

  static Singleton instance = null;


  private Singleton() {
  }

  static Singleton getInstance() {

    if (Singleton.instance == null) {
      Singleton.instance = new Singleton();

    }
    return Singleton.instance;
  }


  public static void main(String[] args) {

    Singleton singleton1 = Singleton.getInstance();
    Singleton singleton2 = Singleton.getInstance();
    Singleton singleton = new Singleton();
    System.out.println("version 01" + singleton1);
    System.out.println("version 02" + singleton2);
    System.out.println("ccc" + singleton);

  }

}


